### ISSUE TITLE
Missing compilation/installation instructions

### ISSUE THREAD

**Issue by rhofour** [timbertson/slinger#1]
I tried installing slinger based on the shellshape installation instructions, but I couldn't get it to work.

Using NVM with NodeJs v6.11.4 I installed typescript. Then I ran:

> tools/gup compile

This gave me:

> gup  compile
> gup    schemas/all
> gup      schemas/inputs
> gup    extension/all
> gup      extension/extension_impl.js
> error TS6053: File 'src/extension_impl.ts' not found.
> gup      Target `extension/extension_impl.js` failed with exit status 2
> gup    Target `extension/all` failed with exit status 2
> gup  Target `compile` failed with exit status 2

Any idea what I'm missing?

**Comment by cookiengineer** [comment]
Yes, same here (on Arch Linux with node v9.4.0 and typescript installed).

For other Arch users: `tsc` aka typescript compiler is necessary, so you need to install `typescript` (from community) first:

```bash
sudo pacman -S nodejs npm typescript;
cd /path/to/slinger;
./tools/gup compile;
```

@timbertson ~~I also tried out all the other branches, but seemingly the `extension_impl.js` file is not commited anywhere.~~ I was mistaken, extension_impl.js was in another folder (see below).

**Comment by timbertson** [comment]
huh, that's werid. `src/extension_impl.ts` is definitely checked in: https://github.com/timbertson/slinger/blob/master/src/extension_impl.ts

I've added a bit of debugging on master branch (c490a4a41451c67503b01539764c8bc69cde6e81) - can you run `tools/gup -x compile` and post the output if it still doesn't work?

**Comment by cookiengineer** [comment]
Ha I think I know what the problem is!

tsc is executed in the wrong directory, it's trying to get into ./src/extension_impl.ts, but the current working directory seems to be inside ./extension.

I modified the `./gup/make-js` file and used `../src` instead of `./src` and it now works fine!

Here's the output of the trace without the fix:

```bash
[✔]─[cookiengineer@tinky]─[/home/cookiengineer/Software/__OTHER/slinger]─[master]
[$] ./tools/gup -x compile
gup  compile
gup   # /home/cookiengineer/Software/__OTHER/slinger
gup   + bash -eu /home/cookiengineer/Software/__OTHER/slinger/compile.gup /home/cookiengineer/Software/__OTHER/slinger/.gup/out.compile compile
gup    schemas/all
gup     # /home/cookiengineer/Software/__OTHER/slinger/schemas
gup     + bash -eu /home/cookiengineer/Software/__OTHER/slinger/schemas/all.gup /home/cookiengineer/Software/__OTHER/slinger/schemas/.gup/out.all all
gup      schemas/inputs
gup       # /home/cookiengineer/Software/__OTHER/slinger/schemas
gup       + bash -eu /home/cookiengineer/Software/__OTHER/slinger/schemas/inputs.gup /home/cookiengineer/Software/__OTHER/slinger/schemas/.gup/out.inputs inputs
gup      schemas/gschemas.compiled
gup       # /home/cookiengineer/Software/__OTHER/slinger/schemas
gup       + bash -eux /home/cookiengineer/Software/__OTHER/slinger/schemas/gschemas.compiled.gup /home/cookiengineer/Software/__OTHER/slinger/schemas/.gup/out.gschemas.compiled gschemas.compiled
+ mkdir -p compiled
+ gup -u inputs
+ glib-compile-schemas . --targetdir=compiled
+ mv compiled/gschemas.compiled /home/cookiengineer/Software/__OTHER/slinger/schemas/.gup/out.gschemas.compiled
+ rm -rf compiled
gup    extension/all
gup     # /home/cookiengineer/Software/__OTHER/slinger/extension
gup     + bash -eu /home/cookiengineer/Software/__OTHER/slinger/gup/extension/all.gup /home/cookiengineer/Software/__OTHER/slinger/extension/.gup/out.all all
gup      extension/extension_impl.js
gup       # /home/cookiengineer/Software/__OTHER/slinger/extension
gup       + /bin/bash /home/cookiengineer/Software/__OTHER/slinger/gup/make-js /home/cookiengineer/Software/__OTHER/slinger/extension/.gup/out.extension_impl.js extension_impl.js
+ pwd
/home/cookiengineer/Software/__OTHER/slinger/extension
+ gup --always
++ basename extension_impl.js .js
+ base=extension_impl
+ '[' -f mode.sh ']'
+ gup --ifcreate mode.sh
+ '[' '' = 1 ']'
+ tsc ./src/extension_impl.ts --alwaysStrict --noImplicitAny --noImplicitReturns --noImplicitThis --noUnusedParameters --noUnusedLocals --out /home/cookiengineer/Software/__OTHER/slinger/extension/.gup/out.extension_impl.js
error TS6053: File 'src/extension_impl.ts' not found.
gup      Target `extension/extension_impl.js` failed with exit status 2
gup    Target `extension/all` failed with exit status 2
gup  Target `compile` failed with exit status 2
```

**Comment by timbertson** [comment]
Ah, thanks! Looks like I had an older globally-installed version of `gup` which had a bug with where it ran some build scripts from. Should be fixed on master now (c552386cba5f1e4c39c95a7319380b351a797f71). I've also added instructions to the readme.

**Comment by cookiengineer** [comment]
Awesome!
